### ISSUE TITLE
[Bug]: WGPU unbroadcast has the same scalar-gradient hole CUDA just fixed

### ISSUE THREAD

**Issue by xupremix** [xupremix/incin#121]
## Affected Component / Subsystem
- incin-backends (WGPU / WebGPU)

## Description of the Bug
`crates/incin-backends/src/wgpu/tape.rs:166` (`unbroadcast`) is line-for-line the same shape as the CUDA version that just failed on hardware: it reduces leading and size-1 axes, then hands anything smaller straight on. A mean reduction seeds the walk with a scalar the next recipe needs at full width. On CUDA this produced `Shape mismatch during 'iteration_plan': expected [], got [3]` for l1-mean backward (fixed on develop with a materialized broadcast after a compatibility check).

Whether it bites on WGPU depends on one fact I could not check here: do the WGPU binary kernels broadcast a scalar operand implicitly (hiding it the way the CPU ones do), or refuse like the CUDA ones? If they refuse, every scalar-reduction backward through a non-broadcast consumer (l1-mean, and any mean(abs(x)) shape) fails the same way.

## Minimal Reproducible Code
```rust
// L1 over three elements, reduced with Mean, then backward.
// Expect: pred gradient [0, -1/3, -1/3] (sign(0) = 0).
// Suspect: iteration_plan shape refusal, as on CUDA before the fix.
let pred = upload([1.0, 0.0, -1.0]);   // WGPU storage, shape [3]
let targ = upload([1.0, 1.0, 0.0]);
let loss = l1_loss(pred, targ, Mean);  // forward: 2/3
let grads = backward(loss);            // <- refusal here?
```
Concretely: build pred/targ as WGPU storage, run the backend L1 composition (sub, abs, mean) or `dispatch::execute`::<op::L1Loss>, then the backend backward, and read the pred gradient.

## Compiler, Error, or Panic Output
Expected (by analogy with CUDA):
```text
Shape mismatch during 'iteration_plan': expected [], got [3]. input shape [3] cannot be normalized to output shape []
```
Unverified on WGPU hardware: filed from code reading plus the proven CUDA twin.

## Environment & Toolchain
rustc 1.92.0, Ubuntu 24.04 x86_64, Incin develop at 2daea9bb. No WGPU hardware here; the software-adapter CI job may reproduce it.

## Confirmation & Checklist
- [X] I have searched existing issues and discussions to confirm this is not a duplicate.
- [ ] I have tested or reproduced this with the latest `develop` or release version.
